rules/data-preprocessing.smk

```diff
@@ -1,35 +1,25 @@
 """This is a Snakemake file defining rules to retrieve raw data from online sources."""
 import pycountry
+import pandas as pd
 
 URL_LOAD = "https://data.open-power-system-data.org/time_series/2018-06-30/time_series_60min_stacked.csv"
 URL_NUTS = "http://ec.europa.eu/eurostat/cache/GISCO/geodatafiles/NUTS_2013_01M_SH.zip"
 URL_LAU = "http://ec.europa.eu/eurostat/cache/GISCO/geodatafiles/COMM-01M-2013-SH.zip"
 URL_DEGURBA = "http://ec.europa.eu/eurostat/cache/GISCO/geodatafiles/DGURBA_2014_SH.zip"
 URL_LAND_COVER = "http://due.esrin.esa.int/files/Globcover2009_V2.3_Global_.zip"
 URL_PROTECTED_AREAS = "https://www.protectedplanet.net/downloads/WDPA_Feb2019?type=shapefile"
-URL_CGIAR_TILE = "http://srtm.csi.cgiar.org/wp-content/uploads/files/srtm_5x5/TIFF/"
-URL_GMTED_TILE = "https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/topo/downloads/GMTED/Global_tiles_GMTED/075darcsec/mea/"
+URL_SRTM_TILE = "http://viewfinderpanoramas.org/dem3/"
```

Comment:
I am uncomfortable with the data source. It's unofficial, and home made. Are the procedures to derive the dataset available?

What about EU-DEM? Would that be an alternative? IT's available in 1 arc second, should cover our scope, and is provided through permalinks.

Reply:
EU-DEM looks perfect actually, yeah! The issue with SRTM was the lack of northern Norway and Iceland data, which EU-DEM covers nicely. I'll update to depend on that.

Reply:
Also, because EU-DEM is given in EPSG:3035, eventually we could (or should?) move this to EPSG:3035. The main reason to _not_ use EPSG:3035 was the land use data set, which we could replace with the European Corine (given in EPSG:3035).

Reply:
Hmm, is there any benefit to consistently using EPSG:3035? It would move us to cartesian coordinates, which would be useful for certain tasks, but is far less easy to understand (e.g. the centroids and bounds make no sense in EPSG:3035).

Reply:
Actually I don't understand your points. Why do centroids and bounds make no sense in EPSG:3035?

One advantage of EPSG:3035 is the consistent cell size. Another advantage is that several (many?most?) data sources actually ARE given in EPSG:3035 (for example population & European Settlement Map, now also EU-DEM), and we are reprojecting them onto WGS84.

In any case, we don't have to decide this here. But I think it's worth keeping in mind.

Reply:
I mean they aren't as readable; specifying coordinates in lat/lon is much more common than in x/y. This is just a matter of taking in lat/lon from config and making sure we convert those to x/y for use in EPSG:3035.
Another issue is that EPSG:3035 is technically not valid for Iceland and Cyprus (they are outside the defined bounds: -10.6700, 34.5000, 31.5500, 71.0500) which suggests distortion is inevitable for those two countries.

Otherwise I think it makes sense to move to that projection if the majority of datasets are consistent with it. Perhaps an issue on it is worth starting, to consolidate these points.

Reply:
> EPSG:3035 is technically not valid for Iceland and Cyprus

Good point, didn't know that!